packages/block-library/src/list-item/edit.native.js

```diff
@@ -116,8 +116,26 @@ export default function ListItemEdit( {
 		} ),
 	};
 
+	const preventDefault = useRef( false );
+	const { onEnter } = useEnter( { content, clientId }, preventDefault );
 	const onSplit = useSplit( clientId );
 	const onMerge = useMerge( clientId );
+	const onSplitList = useCallback(
+		( value ) => {
+			if ( ! preventDefault.current ) {
+				return onSplit( value );
+			}
+		},
+		[ clientId, onSplit ]
+	);
+	const onReplaceList = useCallback(
+		( blocks, ...args ) => {
+			if ( ! preventDefault.current ) {
```

Comment:
What does this do? Why is it different from web?

Reply:
👋🏻 @ellatrix. It looks like this was introduced in https://github.com/WordPress/gutenberg/pull/43949. That may have additional details or be a good location for discussion. Personally, I am not familiar with the details currently. cc/ @geriux

Reply:
> What does this do? Why is it different from web?

We have a different logic when handling "Enter" events on native for RichText components, I remember we had to diverge with our own `useEnter` hook to allow splitting Lists in the middle. I can check the code further to provide more information since we introduced this a little over a year ago.

Is there something that's affecting this code @ellatrix or is it just have more context on this different behavior? Let us know if you want more information about it. Thanks!